src/vinylcontrol/vinylcontrolxwax.cpp

```diff
@@ -171,7 +172,15 @@ VinylControlXwax::VinylControlXwax(UserSettingsPointer pConfig, const QString& g
     // do this once across the VinylControlXwax instances.
     s_xwaxLUTMutex.lock();
 
-    timecoder_init(&timecoder, tc_def, speed, sampleRate.value(), /* phono */ false);
+    bool use_legacy_pitch_filter;
+
+    if (strPitchEstimator == MIXXX_VINYL_PITCH_FILTER_LEGACY) {
+        use_legacy_pitch_filter = true;
+    } else {
+        use_legacy_pitch_filter = false;
+    }
```

Comment:
NIT: This is C++, so I'm gonna be stricter about the style complaints here ;)

Just like in plain C (moreso even in C++), use `const` whenever possible.  
Incidentally that also prevents you from the other grave mistake: declaring variables without initializing them (in this case its trivial so see that its always initialized, but future refactorings may make that harder). So avoid declaring variables without defining them (`const` forces you to do that).

```suggestion
    const bool use_legacy_pitch_filter = strPitchEstimator == MIXXX_VINYL_PITCH_FILTER_LEGACY;
```